src/commands/registerCommands.ts

```diff
@@ -90,14 +93,27 @@ export function registerCommands(
     registerCommandWithTreeNodeUnwrapping('azureFunctions.addBinding', addBinding);
     registerCommandWithTreeNodeUnwrapping(
         'azureFunctions.appSettings.add',
-        async (context: IActionContext, node?: AzExtParentTreeItem) =>
-            await createChildNode(context, new RegExp(AppSettingsTreeItem.contextValue), node),
-    );
+        async (context: ISubscriptionActionContext, node?: AzExtParentTreeItem) => {
+            if ((isResolvedFunctionApp(node?.parent))) {
+                const client: IAppSettingsClient = await node?.parent.site.createClient(context)
```

Comment:
Can you make this a helper function somewhere and just call it? It's a lot of clutter and I think all 3 of them are the same.